src/app/backend/handler/apihandler.go

```diff
@@ -1474,6 +1487,127 @@ func (apiHandler *APIHandler) handleGetPersistentVolumeClaimDetail(request *rest
 	response.WriteHeaderAndEntity(http.StatusOK, result)
 }
 
+type PtyHandler interface {
+	io.Reader
+	io.Writer
+	remotecommand.TerminalSizeQueue
+}
+
+func (apiHandler *APIHandler) handleAttachHelper(request *restful.Request, cmd []string, ptyHandler PtyHandler) error {
+	namespace := request.PathParameter("namespace")
+	podName := request.PathParameter("pod")
+	containerName := request.PathParameter("container")
+
+	req := getApiClient(request).Core().RESTClient().Post().
```

Comment:
As this is related to pod I'd move this logic to i.e. `attach.go` or `terminal.go` file under pod resource. As this is actually exec then maybe we can keep it under some generic name like terminal/console.